### ISSUE TITLE
Feature request: a rule to check whether functions are actually usable within page.evaluate callbacks

### ISSUE THREAD

**Issue by justinfarrelldev** [playwright-community/eslint-plugin-playwright#194]
Often, I find that I'll import functions and use them within page.evaluate calls by accident (then my Playwright tests fail because those functions are not available in the browser context). 

It would be awesome to have a rule that protects against this, though I can definitely see how this may be difficult to implement.

Thank you for the awesome plugin!

**Comment by mskelton** [comment]
@justinfarrelldev I like the idea, my one concern is how this would work with `window` variables. For example:

```js
const x = 10
const result = await page.evaluate(() => {
  return Promise.resolve(x);
}, []);
```

While this looks to be invalid, there isn't any way for ESLint to know that `window.x` does not exist. Yes, this is an edge case for sure, but just thinking out loud here.

Essentially what it sounds like this rule would require is looking at any variables within the `page.evaulate` and ensuring their scope is either in the block, or in the arguments to the `page.evaulate` params.

**Comment by justinfarrelldev** [comment]
> @justinfarrelldev I like the idea, my one concern is how this would work with `window` variables. For example:
> 
> ```js
> const x = 10
> const result = await page.evaluate(() => {
>   return Promise.resolve(x);
> }, []);
> ```
> 
> While this looks to be invalid, there isn't any way for ESLint to know that `window.x` does not exist. Yes, this is an edge case for sure, but just thinking out loud here.

I never really thought about that, but that's a great point. I wonder if there's a way to inspect the window object at runtime? This would certainly slow down the plugin, though...

> Essentially what it sounds like this rule would require is looking at any variables within the `page.evaulate` and ensuring their scope is either in the block, or in the arguments to the `page.evaulate` params.

This sounds promising! Maybe it could be handled similarly to how `eslint-plugin-react` handles missing `useEffect` dependencies (that is, show warnings for missing dependencies - in this case, make sure that the variables are in the params if they are used within `page.evaluate` and not defined in the scope within `page.evaluate` - with a quick fix option in VS Code)?

I'm definitely just spitballing - I'm not familiar even in the slightest with the code in this repo, and I have only been using Playwright for a few months, so forgive me if I haven't made much sense!
